### ISSUE TITLE
GCNet pretrained model?

### ISSUE THREAD

**Issue by TWDH** [open-mmlab/mmdetection#3247]
Hello @hellock , I'm not sure if GCNet in mmdetection use ImageNet pretrained model (specifically GCNet + ResNet pretrained model) cause I saw in the GCNet/config, it only uses standard "torchvision://resnet50" as pretrained model, but in the original paper, the auther firstly trained on ImageNet and transfor to COCO detection task. 

Would you explain does mmdetection use (ResNet pretrained model) or (ResNet+GC pretrained model)? if you only use ResNet pretrained model, why does COCO have such a large improvement?

**Comment by ZwwWayne** [comment]
No. We do not use GCNet pre-trained models. GCBlocks are combined into the common pre-trained ResNet backbone networks during COCO training.

**Comment by TWDH** [comment]
@ZwwWayne Thanks so much for your quick replying~ In my opinion, networks like SENet, CBAM or recently published [ECANet](https://arxiv.org/abs/1910.03151) or [GCTNet](https://openaccess.thecvf.com/content_CVPR_2020/papers/Yang_Gated_Channel_Transformation_for_Visual_Recognition_CVPR_2020_paper.pdf) are all attention-based methods that are pretty much the same as GCNet which generally add a attention module into the resnet residual block. 

However, from my experiment in terms of SENet and CBAM with original pre-trained ResNet backbone, the mAP actually goes down(or no improvement). I asked some of the researchers and they said adding a new module into the pre-trained model will destory the origin  generalization ability and only (for example, SE+ResNet) pre-trained model will show promising results when transforing to  detection task.

I just confused why GCNet can train COCO without GC-ResNet pre-trained model while SENet cannot, or maybe there are finetuning skills I'm not aware of? Looking forward to see your comment! Thanks~

**Comment by xjl-le** [comment]
@TWDH @ZwwWayne  In recent days,I add context_block of gctNet into the lowst feature of FPN,when I train FCOS.but the result is depressed。The average accuracy of model is 29.9 which is lower than average accuracy of original FCOS.Do you konw why?.Looking forward to see your reply。

**Comment by TWDH** [comment]
Hi @xjl-le, Did you add GCNet to the ResNet-Residual block or other places? As far as I know, attention block usually added to the backbone network(like resnet bottleneck) not the FPN part.  I also tried to add GCNet into Centernet(object as point) in the DLA backbone and tested in COCO dataset which increases about 2%.  I don't really know why simply add a attention block into backbone sometimes work and sometimes not... it will be really great if the professional @ZwwWayne can give a brief answer, maybe it's because of destroying backbone pretrained weights or other fancier problems? Thanks

**Comment by xjl-le** [comment]
@TWDH Thanks so much for your quick replying.inspired by Libra R-CNN,I designed the constraction.this result surprised me.but receiving you reply make happy. thanks you

**Comment by TWDH** [comment]
That’s really weird, have you tried origin non-local (or simplified non local)to refine the FCOS?  Theoretically GC should work better than standard non-local

**Comment by xjl-le** [comment]
your suggestion sound good.in the next expriment.i will try origin non-local.thanks for you help

**Comment by TWDH** [comment]
let me know if you have the results, i’m also interest, Thanks~

**Comment by xjl-le** [comment]
OK

**Comment by xvjiarui** [comment]
Hi @TWDH @xjl-le 
Thanks for your question. 
When inserting context block into pretrained backbone, there are a few things you need to pay attention to:
* zero init: If we don't use zero_init, the pretrained weight will be jeopardized by inserted block. 
* Add is better than multiplication for pretrained backbone
* Since the feature statistics will be affected by the inserted block, SyncBN will help to update statistics accordingly. 
* The global context works better with higher-level features (with more semantics).

**Comment by xjl-le** [comment]
Hi @xvjiarui Thanks for your replying.the context_block of gctNet i used is zero init and add method.I can't moderfy the context_block. what's more ,the lower feature of FPN aggregated high feature.so the problem is weird. thanks you

**Comment by TWDH** [comment]
@xvjiarui Such a great explanation, but I got to repeat the question mentioned before about the attention block such as SENet, CBAM, GCT or ECANet. All of them need to be pretrained in the ImageNet(attention module + ResNet) to get better performance. However, I don't have much GPU resources to train the whole backbone again in ImageNet from scratch, I was wondering if there is any work around to directly use those attention block in downstram task like Object detection just as GCNet without specific pretrained model?

Will (zero_init + SycnBN) help SENet as well? Thanks so much for your reply!

**Comment by xvjiarui** [comment]
@xjl-le 
I guess simply add contect_block into FPN is hard to learn. We may need more layers of conv to incorporate global context. But I am not sure about that.

**Comment by xvjiarui** [comment]
@TWDH 
For SENet, it is hard to use zero_init, and also hard to training without pretrained weight. You may refer to [here](https://github.com/open-mmlab/mmclassification/blob/master/docs/model_zoo.md) for SENet pretrained weight.
